File: quartz-energy/src/backend/QuartzEnergy.FracSchedule.Services/Vega/Models/Vehicles/Overviews/VehicleOverview.cs
namespace QuartzEnergy.FracSchedule.Services.Vega.Models.Vehicles.Overviews
{
    using System.ComponentModel.DataAnnotations;

    using QuartzEnergy.Common.DataAnnotations.Attributes.Data;

    public sealed class VehicleOverview
    {
        public VehicleOverview(
            int vehicleId, 
            string maker, 
            string model, 
            string contactName)
        {
            this.VehicleId = vehicleId;
            this.Maker = maker;
            this.Model = model;
            this.ContactName = contactName;
        }

        [IntId]
        public int VehicleId { get; }

        [Required]
        [MaxLength(50)]
        public string Maker { get; }

        [Required]
        [MaxLength(50)]
        public string Model { get; }

        [Required]
        [MaxLength(40)]
        public string ContactName { get; }
    }
}

File: quartz-energy/src/frontend/src/app/vega/models/vehicles/overviews/vehicle-overview.ts
import { NumberIdBusinessModel } from '../../../../common/models/entity/number-id-business-model';

export class VehicleOverview extends NumberIdBusinessModel {
    constructor(
        public id: number,
        public vehicleId: number,
        public maker: string,
        public model: string,
        public contactName: string
    ) {
        super(id);
    }
}
